scripts/burn/kayan.lua

```diff
@@ -0,0 +1,186 @@
+-- kayan: probabalistic
+-- melody discovery
+--
+-- ENC2 = reveal melody
+-- KEY2 = discover new melody
+--
+-- support your local
+-- rainforest
+--
+-- created by @burn
+
+engine.name = 'KarplusRings'
+
+local cs = require 'controlspec'
+
+local sequence = {
+  pos = 0,
+  note_data = {},
+  prob_data = {}
+}
+
+local reveal_level = 35;
+local octave_range = 2;
+local transpose = 2;
+local seq_length = 32;
+local scale_data = {};
+local animation_counter = 0;
+local animation_flipflop = true;
+local pluck_distance = 0;
+
+function init()
+
+  params:add_number("tempo",20,240,48)
+  params:set_action("tempo", function(n)
+    t.time = 15/n
+  end)
+
+  params:add_number("length",4,64,32)
+  params:set_action("length", function(n)
+    seq_length = n;
+  end)
+
+  params:add_number("transpose",0,5,2)
+  params:set_action("transpose", function(n)
+    transpose = n;
+  end)
+
+  params:add_number("octave_range",1,4,2)
+  params:set_action("octave_range", function(n)
+    octave_range = n;
+  end)
+
+  params:add_separator()
+
+  cs.AMP = cs.new(0,1,'lin',0,0.5,'')
+  params:add_control("amp",cs.AMP)
+  params:set_action("amp",
+  function(x) engine.amp(x) end)
+
+  cs.DECAY = cs.new(0.1,15,'lin',0,3.6,'s')
+  params:add_control("damping",cs.DECAY)
+  params:set_action("damping",
+  function(x) engine.decay(x) end)
+
+  cs.COEF = cs.new(0,1,'lin',0,0.11,'')
+  params:add_control("brightness",cs.COEF)
+  params:set_action("brightness",
+  function(x) engine.coef(x) end)
+
+  cs.LPF_FREQ = cs.new(100,10000,'lin',0,4500,'')
+  params:add_control("lpf_freq",cs.LPF_FREQ)
+  params:set_action("lpf_freq",
+  function(x) engine.lpf_freq(x) end)
+
+  cs.LPF_GAIN = cs.new(0,3.2,'lin',0,0.5,'')
+  params:add_control("lpf_gain",cs.LPF_GAIN)
+  params:set_action("lpf_gain",
+  function(x) engine.lpf_gain(x) end)
+
+  cs.BPF_FREQ = cs.new(100,10000,'lin',0,0.5,'')
+  params:add_control("bpf_freq",cs.BPF_FREQ)
+  params:set_action("bpf_freq",
+  function(x) engine.bpf_freq(x) end)
+
+  cs.BPF_RES = cs.new(0,4,'lin',0,0.5,'')
+  params:add_control("bpf_res",cs.BPF_RES)
+  params:set_action("bpf_res",
+  function(x) engine.bpf_res(x) end)
+
+  create_scale_data();
+  create_sequence_data();
+
+  create_playloop_metro()
+  create_animation_metro()
+
+  params:bang()
+end
+
+-- want a custom scale? define it here
+function create_scale_data()
+  scale_data = {0,2,3,5,7,10,12,14,15,17,19,22,24,26,27,29,31,34,36,38,39,41,43,46}
+end
+
+-- create all of the note and probability data on init
+function create_sequence_data()
+  for i=1, 64 do
+    sequence.note_data[i] = scale_data[math.random (1, 24)];
```

Comment:
optimization suggestion (apologies if unwelcome, I'm just so in love with this script!): `sequence.note_data[i] = scale_data[math.random (1, #scale_data)];` would allow for dynamic sequencing based on custom scales that might be shorter or longer than 24 entries.